app/portainer/components/datatables/stacks-datatable/stacksDatatableController.js

```diff
@@ -2,17 +2,24 @@ angular.module('portainer.app').controller('StacksDatatableController', [
   '$scope',
   '$controller',
   'DatatableService',
-  function ($scope, $controller, DatatableService) {
+  'Authentication',
+  function ($scope, $controller, DatatableService, Authentication) {
     angular.extend(this, $controller('GenericDatatableController', { $scope: $scope }));
 
     /**
      * Do not allow external items
      */
     this.allowSelection = function (item) {
-      return !(item.External && item.Type === 2);
+      if (item.External && item.Type === 2) {
+        return false;
+      }
+
+      return !(item.External && !this.isAdmin && !this.isEndpointAdmin);
     };
 
     this.$onInit = function () {
+      this.isAdmin = Authentication.isAdmin();
+      this.isEndpointAdmin = Authentication.hasAuthorizations(['EndpointResourcesAccess']);
```

Comment:
In the `stackController.js` (Stack details view) we are using the `PortainerStackDelete` authorization to ensure the authenticated used has the ability to remove the stack or not. We probably should use the same authorization check here.

Reply:
Ah that means we also need to update the stack details instead.

I've added this check because of the fact that StackDelete do not make any difference between external or internal stack, it's basically delete any stack operation.

Here we have new requirements:

* External: PortainerStackDelete + EndpointResourcesAccess
* Internal: PortainerStackDelete